### PR TITLE
Sensitivity analysis:  support of dangling line contingency

### PR DESCRIPTION
**Pull request by annetill**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

No but we know that the feature is missing.

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

Now, you can simulate the loss of a dangling line in AC and DC sensitivity analysis. TODO: branch tripping does not support dangling lines. 

**What is the current behavior?** *(You can also link to an open issue here)*

It is not supported. 

**What is the new behavior (if this is a feature change)?**

We loose a line and a fictitious bus. 

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

**Review by flo-dup — APPROVED** [review]

File: src/test/java/com/powsybl/openloadflow/sensi/ac/AcSensitivityAnalysisContingenciesTest.java [hunk 1]
```diff
@@ -625,4 +630,64 @@ void testContingencyOnHvdc() {
...
+        SensitivityAnalysisResult result = sensiProvider.run(network, VariantManagerConstants.INITIAL_VARIANT_ID, factorsProvider, contingencies,
+                sensiParameters, LocalComputationManager.getDefault())
+                .join();
+        assertEquals(1, result.getSensitivityValues().size());
+        assertEquals(0.3697, getValue(result, "g1", "l1"), LoadFlowAssert.DELTA_POWER);
+        assertEquals(75.272, getFunctionReference(result, "l1"), LoadFlowAssert.DELTA_POWER);
+        assertEquals(0.3695, getContingencyValue(result, "dl1", "g1", "l1"), LoadFlowAssert.DELTA_POWER);
+        assertEquals(36.794, getContingencyFunctionReference(result, "l1", "dl1"), LoadFlowAssert.DELTA_POWER);
+
+        network.getDanglingLine("dl1").getTerminal().disconnect(); // FIXME: if I comment this line, I don't retrieve the base sensitivity value
```
Comment (annetill) [inline src/test/java/com/powsybl/openloadflow/sensi/ac/AcSensitivityAnalysisContingenciesTest.java:651]: @Hadrien-Godard can you look at my `FIXME` please? I am not able to retrieve the sensitivity value on the base case simply.

File: src/test/java/com/powsybl/openloadflow/sensi/ac/AcSensitivityAnalysisContingenciesTest.java [hunk 2]
```diff
@@ -625,4 +630,64 @@ void testContingencyOnHvdc() {
...
+        SensitivityAnalysisResult result = sensiProvider.run(network, VariantManagerConstants.INITIAL_VARIANT_ID, factorsProvider2, contingencies,
+                sensiParameters, LocalComputationManager.getDefault())
+                .join();
+        assertEquals(1, result.getSensitivityValues().size());
+        assertEquals(-0.3697, getValue(result, "load3", "l1"), LoadFlowAssert.DELTA_POWER);
+        assertEquals(75.336, getFunctionReference(result, "l1"), LoadFlowAssert.DELTA_POWER);
+        assertEquals(-0.3700, getContingencyValue(result, "dl1", "load3", "l1"), LoadFlowAssert.DELTA_POWER);
+        assertEquals(3.0071, getContingencyFunctionReference(result, "l1", "dl1"), LoadFlowAssert.DELTA_POWER);
+
+        network.getDanglingLine("dl1").getTerminal().disconnect(); // FIXME: if I comment this line, I don't retrieve the base sensitivity value
```
Comment (annetill) [inline src/test/java/com/powsybl/openloadflow/sensi/ac/AcSensitivityAnalysisContingenciesTest.java:682]: @Hadrien-Godard Same here.

File: src/test/java/com/powsybl/openloadflow/sensi/ac/AcSensitivityAnalysisContingenciesTest.java (hunk 1 above)
Comment (Hadrien-Godard) [reply src/test/java/com/powsybl/openloadflow/sensi/ac/AcSensitivityAnalysisContingenciesTest.java:651]: Solved with @annetill

File: src/test/java/com/powsybl/openloadflow/sensi/ac/AcSensitivityAnalysisContingenciesTest.java (hunk 2 above)
Comment (Hadrien-Godard) [reply src/test/java/com/powsybl/openloadflow/sensi/ac/AcSensitivityAnalysisContingenciesTest.java:682]: Solved with @annetill

**Review by Hadrien-Godard — APPROVED** [review]

**Pull request merged**